src/js_network_transport.hpp

```diff
@@ -180,30 +180,28 @@ struct JavaScriptNetworkTransport : public JavaScriptNetworkTransportWrapper<T>
         network_transport = Object::get_property(m_ctx, realm_constructor, "_networkTransport");
     };
 
-    void send_request_to_server(const app::Request request, std::function<void(const app::Response)> completion_callback) override {
-        // Create a response handler
-        ObjectType response_handler = ResponseHandlerClass<T>::create_instance(m_ctx, std::move(completion_callback));
-
-        // Create a JavaScript version of the request
-        ObjectType request_object = Object::create_empty(m_ctx);
-        Object::set_property(m_ctx, request_object, "method", Value::from_string(m_ctx, fromHttpMethod(request.method)));
-        Object::set_property(m_ctx, request_object, "url", Value::from_string(m_ctx, request.url));
-        Object::set_property(m_ctx, request_object, "timeoutMs", Value::from_number(m_ctx, request.timeout_ms));
-        if (!request.body.empty()) {
-            Object::set_property(m_ctx, request_object, "body", Value::from_string(m_ctx, request.body));
-        }
-        ObjectType headers_object = Object::create_empty(m_ctx);
+    static ObjectType makeRequest(ContextType ctx, const app::Request& request) {
```

Comment:
This was extracted and cleaned up from the `send_request_to_server` method, so that it could be reused for `watch()` requests which just need to be able to get the POJsO version of the `app::Request`.